tests/structures/visitors/test_required_variables_visitor.py

```diff
@@ -0,0 +1,39 @@
+import pytest
+from decompiler.structures.pseudo import (
+    Assignment,
+    BinaryOperation,
+    Call,
+    DataflowObject,
+    IndirectBranch,
+    Integer,
+    ListOperation,
+    RegisterPair,
+    Return,
+    UnaryOperation,
+    Variable,
+)
+from decompiler.structures.pseudo.operations import OperationType
+from decompiler.structures.visitors.required_variables_visitor import RequiredVariablesVisitor
+
+_a = Variable("a", Integer.int32_t(), 0)
+_b = Variable("b", Integer.int32_t(), 1)
+
+
+@pytest.mark.parametrize(
+    ["obj", "expected_requirements"],
+    [
+        (_a, [_a]),
+        (_r := RegisterPair(_a, _b), [_r, _b, _a]),
+        (Assignment(_a, _b), [_b]),
+        (Assignment(ListOperation([_a]), _b), [_b]),
+        (Assignment(UnaryOperation(OperationType.cast, [_a], contraction=True), _b), [_b]),
+        (Assignment(UnaryOperation(OperationType.dereference, [_a]), _b), [_a, _b]),
+        (IndirectBranch(_a), [_a]),
+        (Return([_a, _b]), [_a, _b]),
+        (ListOperation([_a, _b]), [_a, _b]),
+        (BinaryOperation(OperationType.plus, [_a, _b]), [_a, _b]),
+        (Call(_a, [_b]), [_a, _b])
```

Comment:
Same here, I am missing the test for the case which caused the re-implementation of `requirements` behavior (a variable used more than once in the instruction leads to being present more than once in the `expected_requirements`